libraries/ms-common/src/x86/uart.c

```diff
@@ -136,3 +140,16 @@ void prv_rx_push(UARTPort uart) {
     LOG_DEBUG("Could not read from serial port.");
   }
 }
+
+static void prv_handle_irq(UARTPort uart) {
+  prv_tx_pop(uart);
+  prv_rx_push(uart);
+}
+
+void USART1_IRQHandler(void) {
```

Comment:
Do you understand what these functions are? Your x86 implementation shouldn't need them.

Please don't just copy and paste the STM32 implementation. That module has design decisions that were made because of how the hardware works. The x86 implementation does not require the same behavior.